### ISSUE TITLE
Clarify the different modes of operation

### ISSUE THREAD

**Issue by thusoy** [thusoy/pwm#5]
We're thinking of enabling two different modes of pwm operation.
### Mode 1: Standalone

pwm uses a local SQLite that keeps all the data necessary, and if the user wants to use the application on several devices, it's his/her responsibility to keep the SQLite files in sync. This is the easiest way to get started.
### Mode 2: Backend-powered

To enable automatic synchronization of databases, and to enable access from foreign devices, you need a server with a web interface. Access to the interface must be restricted, so you need a username/password combo here, possibly only password if the server is single-user only. New devices or clients can be set up by only providing the URL to the server, and the client should then generate a CSR and send to the provided URL. When the user has approved the CSR through the web interface, the device will be able to download the certificate and can use that for further communication with the server. The client can thus regularly poll the server for new domains or post new domains that should be synced up. The user then doesn't have to bother with keeping the SQLite databases in sync.

**Comment by afflux** [comment]
Might be worth to consider to have client just generating self-signed certificates and let the server store key fingerprints, instead of implementing a proper PKI. What do you think?

**Comment by thusoy** [comment]
If you can't verify the authority, how do you then prevent anyone from forging a certificate with a given fingerprint?

**Comment by afflux** [comment]
The key fingerprint is supposed to be unique. (But granted, could be bad depending on the hash-function.) You could store the whole list of authorized public keys instead and match against them, but that doesn't really have a clear advantage over rolling your PKI.

**Comment by thusoy** [comment]
Exactly, with self-signed certificates they could opt for MD5 hashes, where finding a collision isn't that hard. But that might not be a problem, the fingerprint is only stored on the server anyway, and isn't publicly available, so it might be a simple way to get started.

**Comment by afflux** [comment]
The certificate hash would be irrelevant, because you'd operate on the key itself. The key fingerprint is computed by a hash function of your choice on the public key. 

I'll have a look into the PKI method today.

**Comment by thusoy** [comment]
Ah, ok, might be feasible then. Cool, I'm busy with some other stuff today, but I can help out later.
